packages/terra-tabs/src/Tabs.jsx

```diff
@@ -0,0 +1,226 @@
+    };
+  }
+
+  getInitialState() {
+    if (this.props.activeKey) {
+      return null;
+    }
+    return TabUtils.initialSelectedTabKey(this.props.children, this.props.defaultActiveKey);
+  }
+
+  getActiveTabIndex() {
+    let activeIndex = -1;
+    React.Children.forEach(this.props.children, (child, index) => {
+      if (child.key === this.state.activeKey || child.key === this.props.activeKey) {
+        activeIndex = index;
+      }
+    });
+
+    return activeIndex;
+  }
+
+
+  handleOnChange(event, selectedPane) {
+    if (!selectedPane.props.isDisabled) {
+      if (this.props.onChange) {
+        this.props.onChange(event, selectedPane.key);
+      } else if (TabUtils.shouldHandleSelection(this.state.activeKey, selectedPane.key)) {
+        this.setState({ activeKey: selectedPane.key });
+      }
+    }
+  }
+
+  handleTruncationChange(isLabelTruncated) {
+    if (this.state.isLabelTruncated !== isLabelTruncated) {
+      this.setState({ isLabelTruncated });
+    }
+  }
+
+  wrapPaneOnClick(pane) {
+    return (event) => {
+      this.handleOnChange(event, pane);
+
+      if (pane.props.onClick) {
+        pane.props.onClick(event);
+      }
+    };
+  }
+
+  render() {
+    const {
+      tabFill,
+      fill,
+      onChange,
+      children,
+      activeKey,
+      defaultActiveKey,
+      ...customProps
+    } = this.props;
+    const attributes = Object.assign({}, customProps);
+
+    // NOTE: Hardcoding variant to structural until descussions have resolved around if we want modular tabs.
+    const variant = 'structural';
+    const tabsClassNames = cx([
+      'tabs-container',
+      { 'tab-fill': tabFill },
+      variant,
+      attributes.className,
+    ]);
+
+    let content = null;
+    let isIconOnly = false;
+    const clonedPanes = [];
+    React.Children.forEach(children, (child) => {
+      let isActive = false;
+      if (child.key === this.state.activeKey || child.key === activeKey) {
+        isActive = true;
+        content = child.props.children;
+      }
+
+      if (child.props.isIconOnly) {
+        isIconOnly = true;
+      }
+
+      clonedPanes.push(React.cloneElement(child, {
+        className: cx({ 'is-active': isActive }),
+        'aria-selected': isActive,
+        onClick: this.wrapPaneOnClick(child),
+      }));
+    });
+
+    content = React.Children.map(content, contentItem => (
+      React.cloneElement(contentItem, { isLabelHidden: isIconOnly || this.state.isLabelTruncated })
+    ));
+
+
+    const collasibleTabs = (
+      <CollapsibleTabs
+        activeKey={activeKey || this.state.activeKey}
+        activeIndex={this.getActiveTabIndex()}
+        onChange={this.handleOnChange}
+        onTruncationChange={this.handleTruncationChange}
+        variant={variant}
+      >
+        {clonedPanes}
+      </CollapsibleTabs>
+    );
+
+    const collapsedTabs = (
+      <CollapsedTabs activeKey={activeKey || this.state.activeKey} onTruncationChange={this.handleTruncationChange}>
```

Comment:
Should this include `onChange` handler similar to the `CollapsibleTabs` above.

Reply:
The onChange was only needed in the `CollapsibleTabs` for the keyboard navigation, since the arrow keys need to change the active tabt. The collapsed tab items don't need it since it doesn't have any special selection logic.